discussions-store/discussions-store-default/src/main/java/org/xwiki/contrib/discussions/store/internal/AbstractDiscussionContextStore.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.discussions.store.internal;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.xwiki.contrib.discussions.DiscussionReferencesSerializer;
+import org.xwiki.contrib.discussions.domain.references.DiscussionContextReference;
+import org.xwiki.contrib.discussions.store.meta.DiscussionContextMetadata;
+import org.xwiki.contrib.discussions.store.meta.DiscussionMetadata;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage;
+import static org.xwiki.query.Query.XWQL;
+
+/**
+ * Abstract implementation for components needing to find discussion context.
+ *
+ * @version $Id$
+ * @since 2.2
+ */
+public abstract class AbstractDiscussionContextStore
```

Comment:
I don't understand why this method is abstract. Couldn't it just be a internal component providing a method to access the discussion context?

Reply:
I extend it in both `DiscussionContextStore` and `DiscussionMetadataContextStore`. It's not really meant to be used independently that's why it's abstract.

Reply:
I'm still not convinced as the children are not specializing anything from the parent. I think component injection is a better way to compose things here in comparison to inheritance. Won't fight for it here anyway as it's all internal ;)